Slider/Assets/Scripts/Map/Ocean/BottleManager.cs

```diff
@@ -0,0 +1,124 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BottleManager : MonoBehaviour
+{
+    // set in the editor
+    public GameObject bottle_prefab; 
+
+    private int turncounter = 0;
+    private GameObject bottle = null;
+    private bool puzzleActive = false;
+    private bool puzzleSolved = false;
```

Comment:
`puzzleSolved` probably needs to be serialized so it's saved between playthroughs, see `SaveSystem.current.SetBool(string, bool)` (and do a `GetBool(str, bool)` in awake)